src/applications/personalization/view-dependents/manage-dependents/tests/unit/ManageDependentsApp.unit.spec.jsx

```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import { expect } from 'chai';
+import userEvent from '@testing-library/user-event';
+import { renderInReduxProvider } from 'platform/testing/unit/react-testing-library-helpers';
+import { removeDependents } from '../../redux/reducers';
+import ManageDependentsApp from '../../containers/ManageDependentsApp';
+import { SCHEMAS } from '../../schemas';
+
+describe('<ManageDependentsApp />', () => {
+  const mockData = {
+    firstName: 'Cindy',
+    lastName: 'See',
+    ssn: '312-243-5634',
+    dateOfBirth: '05/05/1953',
+    relationship: 'Spouse',
+  };
+  it('renders a form', async () => {
+    const initialState = {
+      removeDependents: {
+        dependentsState: {
+          1: {
+            uiSchema: SCHEMAS.Spouse.uiSchema,
+            formSchema: SCHEMAS.Spouse.schema,
+            formData: {},
+          },
+        },
+      },
+    };
+
+    const screen = renderInReduxProvider(
+      <ManageDependentsApp
+        relationship="Spouse"
+        closeFormHandler={() => {}}
+        stateKey={1}
+        userInfo={{
+          fullName: {
+            firstName: mockData.firstName,
+            lastName: mockData.lastName,
+          },
+          dateOfBirth: mockData.dateOfBirth,
+          ssn: mockData.ssn,
+        }}
+      />,
+      {
+        initialState,
+        reducers: { removeDependents },
+      },
+    );
+    expect(await screen.findByText('Remove dependent', { selector: 'button' }))
```

Comment:
two things:

- you might be able to get away with a simple `getBy`, which avoids needing the `await` so it might be a hair faster. At the very least, if you left this `await findBy` in place, then all the following assertions could be done with `getBy`
- I'd suggest using `*ByRole('button', {name: 'Remove dependent'})` just because it ensures the button is accessible. [The docs](https://testing-library.com/docs/queries/bylabeltext#selector) seem to suggest using `selector` only when it's really important what the literal element type is. I think that usually we care about how the user, or especially assistive tech, would access the page.